info.go

```diff
@@ -0,0 +1,113 @@
+package chi
+
+import (
+	"go/parser"
+	"go/token"
+	"os"
+	"path/filepath"
+	"reflect"
+	"runtime"
+	"strings"
+)
+
+type FuncInfo struct {
```

Comment:
as well, remove info.go and move it to docgen/ package, and converge existing logic that is there